crates/adapters/src/controller/mod.rs

```diff
@@ -773,6 +773,10 @@ impl CircuitThread {
 
         self.finish_replaying();
 
+        // Request the first step even if there is no explicit trigger to make sure
+        // the pipeline initializes output snapshots used by ad hoc queries.
+        self.controller.request_step();
```

Comment:
This could be dangerous.
In ddlog there were (recursive) programs that would never converge for empty inputs, but would converge assuming legal inputs.

Reply:
That's really exotic stuff. In any case, this is not a regression, because we used to have an unconditional periodic clock input until recently.